libraries/AP_HAL/CAN.h

```diff
@@ -0,0 +1,233 @@
+/*
+ * Copyright (C) 2015-2016  Intel Corporation. All rights reserved.
+ *
+ * This file is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This file is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include <inttypes.h>
+
+#include "AP_HAL_Namespace.h"
+#include "utility/functor.h"
+
+/*
+ * Pure virtual interface class, abstracting CAN interface
+ */
+class AP_HAL::CAN {
+public:
+    /*  CAN specification has two parts:
+        part A is for the standard format with an 11-bit identifier
+        part B is for the extended format with a 29-bit identifier
+        A CAN device that uses 11-bit identifiers is commonly called CAN 2.0A and a CAN device that uses 29-bit
+        identifiers is commonly called CAN 2.0B.
+     */
+    enum CanBusType {
+        CANBUS_TYPE_UNKNOWN  = 0,
+        CANBUS_TYPE_2A       = 1,
+        CANBUS_TYPE_2B       = 2,
+    };
```

Comment:
Why do you want to restrict the driver to either mode? CAN networks that use both simultaneously are commonplace.